### PR TITLE
2020.11.18 added HL7 V2 message validator to functional testing pipel…

### PR DESCRIPTION
**Pull request by ACMoretxj**
## Description

* Added the HL7 V2 message validator(the nHapi nuget package) to functional testing pipeline;
* Fixed some errors of HL7 V2 messages;

## Testing

All the functional tests and unittests are passed.

### CODE REVIEW

File: data/SampleData/Hl7v2/ADT01-28.hl7 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿MSH|^~\&|ADT1|GOOD HEALTH HOSPITAL|GHH LAB, INC.|GOOD HEALTH HOSPITAL|198808181126|SECURITY|ADT^A01^ADT_A01|MSG00001|P|2.8||
+﻿MSH|^~\&|ADT1|GOOD HEALTH HOSPITAL|GHH LAB, INC.|GOOD HEALTH HOSPITAL|198808181126|SECURITY|ADT^A01^ADT_A01|MSG00001|P|2.5.1||
```
Comment (BoyaWu10) [inline data/SampleData/Hl7v2/ADT01-28.hl7:1]: Are the errors pointed out by the validator?

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests.csproj [hunk 2]
```diff
@@ -8,6 +8,8 @@
   
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
+    <PackageReference Include="nHapi" Version="2.5.0.6" />
+    <PackageReference Include="System.Configuration.ConfigurationManager" Version="5.0.0" />
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests.csproj:12]: Are we using this package in codes?

Comment (ACMoretxj) [reply src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests.csproj:12]: Yes, we use `nHapi` to validate V2 messages, and the `ConfigurationManager` is a dependency of `nHapi`.

File: data/SampleData/Hl7v2/ADT01-28.hl7 (hunk 1 above)
Comment (ACMoretxj) [reply data/SampleData/Hl7v2/ADT01-28.hl7:1]: Yes, I've tried with multiple validators, actually the `2.8` version number can be barely identified. What's worse is that the `2.8` version number will cause infinite loop(function invoking) of `nHapi` and finally throw a `StackOverflow` exception.

File: src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests.csproj [hunk 3]
```diff
@@ -42,4 +42,4 @@
     </None>
   </ItemGroup>
 
-</Project>
+</Project>
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests.csproj:45]: Any changes here?

**Review by BoyaWu10 — APPROVED** [review]

**Pull request merged**